packages/tools/devtools/devtools-view/src/components/data-visualization/EditableView.tsx

```diff
@@ -0,0 +1,364 @@
+/*!
+ * Copyright (c) Microsoft Corporation and contributors. All rights reserved.
+ * Licensed under the MIT License.
+ */
+
+import React from "react";
+
+import {
+	Button,
+	Dropdown,
+	DropdownProps,
+	Input,
+	InputOnChangeData,
+	Option,
+	makeStyles,
+	shorthands,
+	tokens,
+} from "@fluentui/react-components";
+import {
+	DataEdit,
+	EditData,
+	EditType,
+	FluidObjectValueNode,
+} from "@fluid-experimental/devtools-core";
+
+import { useMessageRelay } from "../../MessageRelayContext";
+import { TreeHeader } from "./TreeHeader";
+
+/**
+ * Input to {@link EditableView}
+ */
+export interface EditableViewProps {
+	node: FluidObjectValueNode;
+	containerKey: string;
+	label: string;
+}
+
+/**
+ * Input to EditableComponents
+ */
+interface EditableComponentProps {
+	node: FluidObjectValueNode;
+	editProp: EditState | undefined;
+	setEditProp: React.Dispatch<React.SetStateAction<EditState | undefined>>;
+	submitChange: (data: EditData) => void;
+}
+
+interface EditState {
```

Comment:
Let's be sure to document this and its properties, including what it means for `value` to be undefined.

Reply:
Same problem as below, need to figure out exactly how we want to do this